unity/EditorPlugin/PluginEntryPoint.cs

```diff
@@ -337,60 +353,63 @@ private static void AdviseRefresh(EditorPluginModel model)
       });
     }
 
-    public enum PlayModeState
-    {
-      Stopped,
-      Playing,
-      Paused
-    }
-
-    public static PlayModeState PlayModeSavedState = PlayModeState.Stopped;
-
     private static void AdviseUnityActions(EditorPluginModel model, Lifetime connectionLifetime)
     {
-      var isPlayingAction = new Action(() =>
+      var syncPlayState = new Action(() =>
       {
         MainThreadDispatcher.Instance.Queue(() =>
         {
-          var isPlayOrWillChange = EditorApplication.isPlayingOrWillChangePlaymode;
-          var isPlaying = isPlayOrWillChange && EditorApplication.isPlaying;
+          var isPlaying = EditorApplication.isPlayingOrWillChangePlaymode && EditorApplication.isPlaying;
           if (!model.Play.HasValue() || model.Play.HasValue() && model.Play.Value != isPlaying)
+          {
+            ourLogger.Verbose("Reporting play mode change to model: {0}", isPlaying);
             model.Play.SetValue(isPlaying);
+          }
 
           var isPaused = EditorApplication.isPaused;
           if (!model.Pause.HasValue() || model.Pause.HasValue() && model.Pause.Value != isPaused)
+          {
+            ourLogger.Verbose("Reporting pause mode change to model: {0}", isPaused);
             model.Pause.SetValue(isPaused);
+          }
         });
       });
-      isPlayingAction(); // get Unity state
-      model.Play.AdviseNotNull(connectionLifetime, play =>
+
+      syncPlayState();
+
+      model.Play.Advise(connectionLifetime, play =>
```

Comment:
What is the reason for changing AdviseNotNull to Advise here?

Reply:
Thanks for clarification.

Reply:
Approved. Anyway, I guess we should add `where` clause to this method to suppress boxing.